packages/contracts-bedrock/contracts/test/SystemConfig.t.sol

```diff
@@ -201,4 +226,37 @@ contract SystemConfig_Setters_Test is SystemConfig_Init {
         sysConf.setUnsafeBlockSigner(newUnsafeSigner);
         assertEq(sysConf.unsafeBlockSigner(), newUnsafeSigner);
     }
+
+    function testFuzz_setSigner_succeeds(address _signer, bool _authenticated) external {
+        address[] memory cachedSigners = sysConf.signerSet();
+        uint256 len = cachedSigners.length;
+        for (uint256 i = 0; i < len; i++) {
+            if (cachedSigners[i] == _signer) {
+                if (_authenticated) {
+                    vm.expectRevert("SystemConfig: signer already authenticated");
+                } else {
+                    vm.expectEmit(true, true, true, true);
+                    emit ConfigUpdate(
+                        0,
+                        SystemConfig.UpdateType.SIGNER_SET,
+                        abi.encode(_signer, _authenticated)
+                    );
+                }
+            }
+        }
+        vm.prank(sysConf.owner());
+        sysConf.setSigner(_signer, _authenticated);
```

Comment:
It would be nice to assert on the post return value of `signerSet` someplace, I don't see thorough test coverage of adding/removing with specific assertions on the signer set. Basically to cover the removal case with the inplace swap of the last element + the pop